src/main/java/com/uid2/shared/auth/AuthorizableStore.java

```diff
@@ -0,0 +1,75 @@
+package com.uid2.shared.auth;
+
+import com.uid2.shared.secret.KeyHasher;
+
+import java.nio.ByteBuffer;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+
+public class AuthorizableStore<T extends IAuthorizable> {
+    private class AuthorizableStoreSnapshot {
+        private final Map<ByteBuffer, T> authorizables;
```

Comment:
Can we use `byte[]` instead as the key?

Reply:
byte[] doesn't support proper hashcode, ByteBuffer does apparently

Reply:
Right, makes sense

I think as long as we don't modify the keys directly (which we shouldn't be doing anyway), this should be fine